src/scripts/main.js

```diff
@@ -1,3 +1,178 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+function formatSalary(cellValue) {
+  return parseFloat(cellValue.replace(/[$,]/g, ''));
+}
+
+function sortRows(index, isAscending, rowsArr) {
+  return rowsArr.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    if (index === 4) {
+      const salaryA = formatSalary(cellA);
+      const salaryB = formatSalary(cellB);
+
+      return isAscending ? salaryA - salaryB : salaryB - salaryA;
+    } else {
+      return isAscending
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    }
+  });
+}
+
+function updateTable(sortedRows) {
+  const tbody = document.querySelector('tbody');
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+}
+
+function handleSortAndUpdate(index, arraySort, directionSort) {
```

Comment:
The function `handleSortAndUpdate` is not pure because it's manipulating the `directionSort` object. It would be better to return a new object with the updated sort direction.